### PR TITLE
Add missing output fields to Certificate resource.

### PR DESCRIPTION
**Pull request by pawel-grz-krawczyk**
Updated a couple of mistakes in fields descriptions. Changed type of
Certificate.Managed.State from String to Enum.

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
certificatemanager: added `state`, `authorization_attempt_info` and `provisioning_issue` output fields to `google_certificate_manager_certificate`
```

### CODE REVIEW

**Comment by pawel-grz-krawczyk** [comment]
These 3 test failures seems to be unrelated to the changes in this PR. Two of them have following error message:

```
Error: Error waiting to create CertificateAuthority: Error waiting for Creating CertificateAuthority: Error code 9, message: Tenancy unit reached the maximum number of tenant projects.
```

and the third:

```
  | Error: Error waiting for starting instance: Instance could not be scheduled due to no matching node with property compatibility.
  | 
  | Explanation:
  | The matching node(s) <tf-test-nodegroup-aylek6rlyi-fwsd> are not in READY status.
  | 
  | Potential fix:
  | - Please consider waiting for the matching node(s) to become READY.
  | - Please consider deleting and recreating the node or contacting Google Cloud Support, in case it is taking several minutes for the node to become READY.
  | 
  | 
  |   with google_compute_instance.foobar,
  |   on terraform_plugin_test.tf line 7, in resource "google_compute_instance" "foobar":
  |    7: resource "google_compute_instance" "foobar" {
  ```

**Comment by shuyama1** [comment]
> These 3 test failures seems to be unrelated to the changes in this PR. Two of them have following error message:
> 
> ```
> Error: Error waiting to create CertificateAuthority: Error waiting for Creating CertificateAuthority: Error code 9, message: Tenancy unit reached the maximum number of tenant projects.
> ```
> 
> and the third:
> 
> ```
>   | Error: Error waiting for starting instance: Instance could not be scheduled due to no matching node with property compatibility.
>   | 
>   | Explanation:
>   | The matching node(s) <tf-test-nodegroup-aylek6rlyi-fwsd> are not in READY status.
>   | 
>   | Potential fix:
>   | - Please consider waiting for the matching node(s) to become READY.
>   | - Please consider deleting and recreating the node or contacting Google Cloud Support, in case it is taking several minutes for the node to become READY.
>   | 
>   | 
>   |   with google_compute_instance.foobar,
>   |   on terraform_plugin_test.tf line 7, in resource "" "foobar":
>   |    7: resource "google_compute_instance" "foobar" {
> ```

Thanks for making the changes! The failed tests are unrelated to this PR. Don't worry about those. Sorry for the noise.    
Got a question about changing the type from `String` to `Enum`. Is there any particular reason that we want to change the type to `Enum`? Are there any issues caused by setting `Certificate.Managed.State` as `String`?
We tend to convert `Enum` type to `String` in our codebase as to avoid any further modification to the field when API add new values in the future.

**Comment by pawel-grz-krawczyk** [comment]
> Is there any particular reason that we want to change the type to Enum? Are there any issues caused by setting Certificate.Managed.State as String?
> We tend to convert Enum type to String in our codebase as to avoid any further modification to the field when API add new values in the future.
There are no issues with the type being String. I just saw a couple of Enums in the api.yaml and noticed that this one field is the only exception. I will revert it back to String

I could also change other Enums to String in this file. In that case I assume that description field for such String type would contains only general description, without details about specific "enum" types. E.g.:

```
                  Reason for provisioning failures.

                  The reason is undefined.

                  Certificate provisioning failed due to an issue with one or more of
                  the domains on the certificate.
                  For details of which domains failed, consult the
                  `authorization_attempt_info` field.

                  Exceeded Certificate Authority quotas or internal rate limits of the
                  system. Provisioning may take longer to complete.
```
Would change to:

```
                  Reason for provisioning failures.
```                  

Is that correct?

File: mmv1/products/certificatemanager/api.yaml [hunk 1]
```diff
@@ -67,9 +67,23 @@ objects:
...
+          Creation timestamp of a DNS Authorization. Timestamp in RFC3339 UTC "Zulu" format,
+          with nanosecond resolution and up to nine fractional digits.
+          Examples: "2014-10-02T15:01:23Z" and "2014-10-02T15:01:23.045123456Z".
+      - !ruby/object:Api::Type::Time
+        name: 'updateTime'
+        description: |
+          Update timestamp of a DNS Authorization. Timestamp in RFC3339 UTC "Zulu" format,
+          with nanosecond resolution and up to nine fractional digits.
+          Examples: "2014-10-02T15:01:23Z" and "2014-10-02T15:01:23.045123456Z".
+        output: true
```
Comment (shuyama1) [inline mmv1/products/certificatemanager/api.yaml:83]: Are these two needed for any particular reason? Like are there any other resources that depend on these?

File: mmv1/products/certificatemanager/api.yaml [hunk 2]
```diff
@@ -141,9 +156,23 @@ objects:
...
+          Creation timestamp of a Certificate. Timestamp in RFC3339 UTC "Zulu" format,
+          with nanosecond resolution and up to nine fractional digits.
+          Examples: "2014-10-02T15:01:23Z" and "2014-10-02T15:01:23.045123456Z".
+      - !ruby/object:Api::Type::Time
+        name: 'updateTime'
+        description: |
+          Update timestamp of a Certificate. Timestamp in RFC3339 UTC "Zulu" format,
+          with nanosecond resolution and up to nine fractional digits.
+          Examples: "2014-10-02T15:01:23Z" and "2014-10-02T15:01:23.045123456Z".
+        output: true
```
Comment (shuyama1) [inline mmv1/products/certificatemanager/api.yaml:172]: same as above

**Review by shuyama1 — COMMENTED** [review]
Thanks for adding the fields and updating the documentation! 
Just left some review comments and also it would nice if we could change all enum type to string. Thanks!

File: mmv1/products/certificatemanager/api.yaml (hunk 1 above)
Comment (pawel-grz-krawczyk) [reply mmv1/products/certificatemanager/api.yaml]: delete updateTime and createTime in all reasources

File: mmv1/products/certificatemanager/api.yaml (hunk 2 above)
Comment (pawel-grz-krawczyk) [reply mmv1/products/certificatemanager/api.yaml]: delete updateTime and createTime in all reasources

File: mmv1/products/certificatemanager/api.yaml [hunk 3]
```diff
@@ -253,20 +292,6 @@ objects:
...
-          Creation timestamp of a Certificate Map. Timestamp is in RFC3339 UTC "Zulu" format,
-          accurate to nanoseconds with up to nine fractional digits.
-          Examples: "2014-10-02T15:01:23Z" and "2014-10-02T15:01:23.045123456Z".
-      - !ruby/object:Api::Type::Time
-        name: 'updateTime'
-        description: |
-          Update timestamp of a Certificate Map. Timestamp is in RFC3339 UTC "Zulu" format,
-          accurate to nanoseconds with up to nine fractional digits.
-          Examples: "2014-10-02T15:01:23Z" and "2014-10-02T15:01:23.045123456Z".
-        output: true
```
Comment (shuyama1) [inline mmv1/products/certificatemanager/api.yaml:269]: Would you mind if we keep the already implemented `createTime` and `updateTime` fields in the providers. Honestly removing those fields is likely not going to affect any users but removing a field is considered a breaking change in general which we can not release in minor release. Just want to be extra cautious and don't want to break any existing users.

File: mmv1/products/certificatemanager/api.yaml [hunk 4]
```diff
@@ -356,20 +381,6 @@ objects:
...
-            Creation timestamp of a Certificate Map Entry. Timestamp in RFC3339 UTC "Zulu" format, 
-            with nanosecond resolution and up to nine fractional digits. 
-            Examples: "2014-10-02T15:01:23Z" and "2014-10-02T15:01:23.045123456Z".
-        - !ruby/object:Api::Type::Time
-          name: 'updateTime'
-          description: |
-            Update timestamp of a Certificate Map Entry. Timestamp in RFC3339 UTC "Zulu" format, 
-            with nanosecond resolution and up to nine fractional digits. 
-            Examples: "2014-10-02T15:01:23Z" and "2014-10-02T15:01:23.045123456Z".
-          output: true
```
Comment (shuyama1) [inline mmv1/products/certificatemanager/api.yaml:372]: same as above. Thanks!

**Review by shuyama1 — CHANGES_REQUESTED** [review]

**Comment by pawel-grz-krawczyk** [comment]
Failed tests seem unrelated to these changes:
- timeouts / deadline exceeded
- missing resource - RESOURCE_NOT_READY error (compute API, instance resource)
- [bad request error](https://00f74ba44b8e08521c5f0024a9d6bebf66e82ff248-apidata.googleusercontent.com/download/storage/v1/b/ci-vcr-logs/o/beta%2Frefs%2Fheads%2Fauto-pr-6322%2Fartifacts%2F72db0cb9-9ab3-4503-9070-8157d5178c5f%2Frecording%2FTestAccCloudRunService_cloudRunServiceStaticOutboundExample.log?jk=AFshE3VRCPLhZhxLvC_4IRuWSMq6ZA9lpAHFVG288xioMNiZ2u2lIOqEz3-hohlI1v2Njt2z8Xaw1_A9Fhq9bS7xctvaH5gKxSa9lqKKL5sbnlaeqJF8STkNb3OCHIxJtFmPJ0HM50361oD7BykP4_97HndqHixYRXLKuGmcNL2YCK-3zWzGVaFU300ck46ZW5XNIH3iBSsx5DL0o-t4Gf4O131Sb49uvXcIc3tJx24a0SpNZKBXyLgMdf1OQlJ4L3PMtZoYpfr9imrv3nTBZJfqJLwS5FqhlyTQv8ym4rfdWGaDW5AovJm6pDdxvuXRkv4PWuSh_vLCMX9axVNF3V4RZqdyO04O3Bh-hXSP_Kmc83C9lhYVqnEL5oU7bRc1FFim1A0M0DbikdhbklrWv7z30zmCG_1ooj0HZdzveiPkfrEkskK928ouZ22x-00gob_PsiqxGGM7JNXSHuL9Bd82IHRjlDL06_Gt8etNGwrlLcKHJFfQK1ZZxkaxPyglY86aztkwYGIgnTvvxzqETkFr7WJvTqkIQpl8V7Cf4hEkDoOMgmYqmKot5dFR7phrXye22mP9DjfrlGRrP_QxFqOhwYvTXFkSXvqpELKBtbSJbUeFVVw4Lo0CYUAWw75_HftWHv69_z2PyGmwveMnzatWq_e-Emzlrd83nr-eA_flQ_7DnAU-tQxN0LxgaSZ_fJ8Kzg8oBYLpHZgovb17e-tWEei2JVZvkw6cFtE2e-0kvubZxUreFDYFNzYC5ZQpFqaJpHsFGcvJU_oNwTjTxUQTFgaUw98g0_1jTBi5AgeSKQ_RRZFUJ-sVifBAlzCbYsuDw-BlZi6Vn7hEc6pD9ZOlVGcYUDlcOXezAX-pGtzDW3ylBhVCh_wtaarLUFXxV9kP78794CfOTSs6oU3TSrFFCts5kbdVGyN8ffRjgXDPzO4igYwBc1d2ZXEQqShypP30nyGeMo_e1QrAqI49xUAZwRdahJkmnvJajmGeI1oac6pGRCnmg2H-APf3LvojpmxyrfPfASg0tU5sAN27DennKHAXb1iVzkH-mvaAIEAGjVh89uow2wv94toVQ8VGL6w2iYAGKhCvv-RoSUYHCaRqeCMKuN5p7WtJmUGmTkIztF7h2l4FUXsFXwDgsW7UPoF3nDJuZ0Y0FKenPX4HA3Z_MUSXChiCYdbg-G7JUiJG4d4DYOn6AGeAZy4XuFtu9PP5M7YnCMWf5s-RXWSg9Q&isca=1) - looks suspicious but unrelated, logs:
```
  | Error creating Service: googleapi: Error 400: metadata.annotations: Annotation 'run.googleapis.com/vpc-access-connector' is not supported on resources of kind 'Service'. Supported kinds are: Revision, Execution
  | Details:
  | [
  |   {
  |     "@type": "type.googleapis.com/google.rpc.BadRequest",
  |     "fieldViolations": [
  |       {
  |         "description": "Annotation 'run.googleapis.com/vpc-access-connector' is not supported on resources of kind 'Service'. Supported kinds are: Revision, Execution",
  |         "field": "metadata.annotations"
  |       }
  |     ]
  |   }
  | ]
  
  ```

**Review by shuyama1 — APPROVED** [review]

**Pull request merged**